### ISSUE TITLE
Beam direction influences slur direction

### ISSUE THREAD

**Issue by craigsapp** [rism-ch/verovio#658]
When beams are forced down in single-layer mode, a slur attached to the notes in that beam is also forced down.  The slur direction should be independent of the forced direction of the beam in such cases.

In the following example, beat 1 is displayed correctly, and there are no forced directions.  

Beat 2 contains a beam forced downwards.  The slur is also positioned below, but it should be placed above by default in this situation (it should go on the notehead side of the notes since all of the stems are in the same direction).

Measure 2, beat 1 is correct.  In this case the beam is forced above, and the slur is positioned below by default.   Measure 2, beat 2 is correct since both the beam direction is forced below and the slur direction is forced above.  But in this case it should not be necessary to force the slur above.

<img width="724" alt="screen shot 2017-07-14 at 10 09 41 pm" src="https://user-images.githubusercontent.com/3487289/28229057-8726417e-68e1-11e7-9697-d5f3cf4183ff.png">

Humdrum data:

```
**kern
*clefG2
*k[f#c#g#]
*M6/8
(8f#L
8cc#)
8a'J
(8f#L<
8cc#)
8a'J
=2
(8f#L>
8cc#)
8a'J
(>8f#L<
8cc#)
8a'J
=
*-
!!!RDF**kern: > = above
!!!RDF**kern: < = below
```

MEI transcoding:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2017-07-14T22:19:08" version="2.0.0-dev-06e4fee">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-0000000018764039">
                        <staffGrp xml:id="staffgrp-0000000634101773">
                            <staffDef xml:id="staffdef-0000000626972708" clef.shape="G" clef.line="2" key.sig="3s" meter.count="6" meter.unit="8" n="1" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-0000001986380150">
                        <measure xml:id="measure-L1" n="0" type="m-0">
                            <staff xml:id="staff-0000000734403237" n="1">
                                <layer xml:id="layer-L1F1N1" n="1">
                                    <beam xml:id="beam-L5F1-L7F1">
                                        <note xml:id="note-L5F1" type="qon-0 qoff-1_2 pname-f acc-s oct-4 b40c-20 b12c-6 " dur="8" oct="4" pname="f" accid.ges="s" />
                                        <note xml:id="note-L6F1" type="qon-1_2 qoff-1 pname-c acc-s oct-5 b40c-3 b12c-1 " dur="8" oct="5" pname="c" accid.ges="s" />
                                        <note xml:id="note-L7F1" type="qon-1 qoff-3_2 pname-a acc-n oct-4 b40c-31 b12c-9 " dur="8" oct="4" pname="a" accid.ges="n">
                                            <artic xml:id="artic-L7F1" artic="stacc" />
                                        </note>
                                    </beam>
                                    <beam xml:id="beam-L8F1-L10F1">
                                        <note xml:id="note-L8F1" type="qon-3_2 qoff-2 pname-f acc-s oct-4 b40c-20 b12c-6 " dur="8" oct="4" pname="f" stem.dir="down" accid.ges="s" />
                                        <note xml:id="note-L9F1" type="qon-2 qoff-5_2 pname-c acc-s oct-5 b40c-3 b12c-1 " dur="8" oct="5" pname="c" stem.dir="down" accid.ges="s" />
                                        <note xml:id="note-L10F1" type="qon-5_2 qoff-3 pname-a acc-n oct-4 b40c-31 b12c-9 " dur="8" oct="4" pname="a" stem.dir="down" accid.ges="n">
                                            <artic xml:id="artic-L10F1" artic="stacc" />
                                        </note>
                                    </beam>
                                </layer>
                            </staff>
                            <slur xml:id="slur-L5F1-L6F1N1" staff="1" startid="#note-L5F1" endid="#note-L6F1" />
                            <slur xml:id="slur-L8F1-L9F1N1" staff="1" startid="#note-L8F1" endid="#note-L9F1" />
                        </measure>
                        <measure xml:id="measure-L11" n="2" type="m-2">
                            <staff xml:id="staff-L11F1N1" n="1">
                                <layer xml:id="layer-L11F1N1" n="1">
                                    <beam xml:id="beam-L12F1-L14F1">
                                        <note xml:id="note-L12F1" type="qon-3 qoff-7_2 pname-f acc-s oct-4 b40c-20 b12c-6 " dur="8" oct="4" pname="f" stem.dir="up" accid.ges="s" />
                                        <note xml:id="note-L13F1" type="qon-7_2 qoff-4 pname-c acc-s oct-5 b40c-3 b12c-1 " dur="8" oct="5" pname="c" stem.dir="up" accid.ges="s" />
                                        <note xml:id="note-L14F1" type="qon-4 qoff-9_2 pname-a acc-n oct-4 b40c-31 b12c-9 " dur="8" oct="4" pname="a" stem.dir="up" accid.ges="n">
                                            <artic xml:id="artic-L14F1" artic="stacc" />
                                        </note>
                                    </beam>
                                    <beam xml:id="beam-L15F1-L17F1">
                                        <note xml:id="note-L15F1" type="qon-9_2 qoff-5 pname-f acc-s oct-4 b40c-20 b12c-6 " dur="8" oct="4" pname="f" stem.dir="down" accid.ges="s" />
                                        <note xml:id="note-L16F1" type="qon-5 qoff-11_2 pname-c acc-s oct-5 b40c-3 b12c-1 " dur="8" oct="5" pname="c" stem.dir="down" accid.ges="s" />
                                        <note xml:id="note-L17F1" type="qon-11_2 qoff-6 pname-a acc-n oct-4 b40c-31 b12c-9 " dur="8" oct="4" pname="a" stem.dir="down" accid.ges="n">
                                            <artic xml:id="artic-L17F1" artic="stacc" />
                                        </note>
                                    </beam>
                                </layer>
                            </staff>
                            <slur xml:id="slur-L12F1-L13F1N1" staff="1" startid="#note-L12F1" endid="#note-L13F1" />
                            <slur xml:id="slur-L15F1-L16F1N1" staff="1" startid="#note-L15F1" endid="#note-L16F1" curvedir="above" />
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by craigsapp** [comment]
Or maybe not directly related to beams.  Here is an example:

<img width="996" alt="screen shot 2017-07-17 at 1 41 03 am" src="https://user-images.githubusercontent.com/3487289/28252326-21b80f3c-6a91-11e7-936b-41ce0b79dbcc.png">
 
The slur in the first measure should be above the notes since the stems all point in the same direction downwards.  The other measures' slurs are correct.

Example data:

```
**kern
(4b
4cc
4dd
4ee)
=
(4B
4c
4d
4e)
=
(4B
4c
4dd
4ee)
=
(4b
4cc
4d
4e)
=
*-
```

**Comment by craigsapp** [comment]
In this example, the first three measures are incorrect, then the next three are correct, and the last one is incorrect:

<img width="954" alt="screen shot 2017-07-17 at 1 50 44 am" src="https://user-images.githubusercontent.com/3487289/28252402-6cc3405e-6a92-11e7-944f-c510e36e8489.png">

```
**kern
(4g\
4a\
4b
4cc)
=2
(4a\
4b
4cc
4dd)
=3
(4b
4cc
4dd
4ee)
=4
(4cc
4dd
4ee
4ff)
=5
(4dd
4ee
4ff
4gg)
=6
(4ee
4ff
4gg
4aa)
=
(4ff/
4gg/
4aa/
4bb/)
=
*-
```
 MEI data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2017-07-17T01:52:29" version="2.0.0-dev-6af5e40">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workDesc>
            <work>
                <titleStmt>
                    <title />
                </titleStmt>
            </work>
        </workDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-0000001580261972">
                        <staffGrp xml:id="staffgrp-0000001986613682">
                            <staffDef xml:id="staffdef-0000001855065792" clef.shape="G" clef.line="2" n="1" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-0000000716509398">
                        <measure xml:id="measure-L1" n="0">
                            <staff xml:id="staff-0000001083831422" n="1">
                                <layer xml:id="layer-L1F1N1" n="1">
                                    <note xml:id="note-L2F1" dur="4" oct="4" pname="g" stem.dir="down" accid.ges="n" />
                                    <note xml:id="note-L3F1" dur="4" oct="4" pname="a" stem.dir="down" accid.ges="n" />
                                    <note xml:id="note-L4F1" dur="4" oct="4" pname="b" accid.ges="n" />
                                    <note xml:id="note-L5F1" dur="4" oct="5" pname="c" accid.ges="n" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-L2F1-L5F1N1" staff="1" startid="#note-L2F1" endid="#note-L5F1" />
                        </measure>
                        <measure xml:id="measure-L6" n="2">
                            <staff xml:id="staff-L6F1N1" n="1">
                                <layer xml:id="layer-L6F1N1" n="1">
                                    <note xml:id="note-L7F1" dur="4" oct="4" pname="a" stem.dir="down" accid.ges="n" />
                                    <note xml:id="note-L8F1" dur="4" oct="4" pname="b" accid.ges="n" />
                                    <note xml:id="note-L9F1" dur="4" oct="5" pname="c" accid.ges="n" />
                                    <note xml:id="note-L10F1" dur="4" oct="5" pname="d" accid.ges="n" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-L7F1-L10F1N1" staff="1" startid="#note-L7F1" endid="#note-L10F1" />
                        </measure>
                        <measure xml:id="measure-L11" n="3">
                            <staff xml:id="staff-L11F1N1" n="1">
                                <layer xml:id="layer-L11F1N1" n="1">
                                    <note xml:id="note-L12F1" dur="4" oct="4" pname="b" accid.ges="n" />
                                    <note xml:id="note-L13F1" dur="4" oct="5" pname="c" accid.ges="n" />
                                    <note xml:id="note-L14F1" dur="4" oct="5" pname="d" accid.ges="n" />
                                    <note xml:id="note-L15F1" dur="4" oct="5" pname="e" accid.ges="n" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-L12F1-L15F1N1" staff="1" startid="#note-L12F1" endid="#note-L15F1" />
                        </measure>
                        <measure xml:id="measure-L16" n="4">
                            <staff xml:id="staff-L16F1N1" n="1">
                                <layer xml:id="layer-L16F1N1" n="1">
                                    <note xml:id="note-L17F1" dur="4" oct="5" pname="c" accid.ges="n" />
                                    <note xml:id="note-L18F1" dur="4" oct="5" pname="d" accid.ges="n" />
                                    <note xml:id="note-L19F1" dur="4" oct="5" pname="e" accid.ges="n" />
                                    <note xml:id="note-L20F1" dur="4" oct="5" pname="f" accid.ges="n" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-L17F1-L20F1N1" staff="1" startid="#note-L17F1" endid="#note-L20F1" />
                        </measure>
                        <measure xml:id="measure-L21" n="5">
                            <staff xml:id="staff-L21F1N1" n="1">
                                <layer xml:id="layer-L21F1N1" n="1">
                                    <note xml:id="note-L22F1" dur="4" oct="5" pname="d" accid.ges="n" />
                                    <note xml:id="note-L23F1" dur="4" oct="5" pname="e" accid.ges="n" />
                                    <note xml:id="note-L24F1" dur="4" oct="5" pname="f" accid.ges="n" />
                                    <note xml:id="note-L25F1" dur="4" oct="5" pname="g" accid.ges="n" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-L22F1-L25F1N1" staff="1" startid="#note-L22F1" endid="#note-L25F1" />
                        </measure>
                        <measure xml:id="measure-L26" n="6">
                            <staff xml:id="staff-L26F1N1" n="1">
                                <layer xml:id="layer-L26F1N1" n="1">
                                    <note xml:id="note-L27F1" dur="4" oct="5" pname="e" accid.ges="n" />
                                    <note xml:id="note-L28F1" dur="4" oct="5" pname="f" accid.ges="n" />
                                    <note xml:id="note-L29F1" dur="4" oct="5" pname="g" accid.ges="n" />
                                    <note xml:id="note-L30F1" dur="4" oct="5" pname="a" accid.ges="n" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-L27F1-L30F1N1" staff="1" startid="#note-L27F1" endid="#note-L30F1" />
                        </measure>
                        <measure xml:id="measure-L31">
                            <staff xml:id="staff-L31F1N1" n="1">
                                <layer xml:id="layer-L31F1N1" n="1">
                                    <note xml:id="note-L32F1" dur="4" oct="5" pname="f" stem.dir="up" accid.ges="n" />
                                    <note xml:id="note-L33F1" dur="4" oct="5" pname="g" stem.dir="up" accid.ges="n" />
                                    <note xml:id="note-L34F1" dur="4" oct="5" pname="a" stem.dir="up" accid.ges="n" />
                                    <note xml:id="note-L35F1" dur="4" oct="5" pname="b" stem.dir="up" accid.ges="n" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-L32F1-L35F1N1" staff="1" startid="#note-L32F1" endid="#note-L35F1" />
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by lpugin** [comment]
Fixed 54e351eabe66c5f967b90ebde03bc7bb465e1f3c

**Comment by craigsapp** [comment]
Looks great:

<img width="847" alt="screen shot 2017-07-17 at 12 14 41 pm" src="https://user-images.githubusercontent.com/3487289/28264098-ac3cc00a-6ae9-11e7-99dd-c8f26401cd8e.png">
<img width="1007" alt="screen shot 2017-07-17 at 12 15 36 pm" src="https://user-images.githubusercontent.com/3487289/28264102-aecc565a-6ae9-11e7-8c56-ba14a2525cb5.png">
<img width="1038" alt="screen shot 2017-07-17 at 12 14 08 pm" src="https://user-images.githubusercontent.com/3487289/28264105-b0a78f26-6ae9-11e7-9b3a-9026d0c9175b.png">
